File: algorithm/2_Add_Two_Numbers/Add_Two_Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        carry = 0
        result = ListNode(0)
        current = result
        while l1 is not None and l2 is not None:
            sum = l1.val + l2.val + carry
            carry = sum // 10
            sum = sum % 10
            current.next = ListNode(sum)
            current = current.next
            l1 = l1.next
            l2 = l2.next
        while l1 is not None:
            sum = l1.val + carry
            carry = sum // 10
            sum = sum % 10
            current.next = ListNode(sum)
            current = current.next
            l1 = l1.next
        while l2 is not None:
            sum = l2.val + carry
            carry = sum // 10
            sum = sum % 10
            current.next = ListNode(sum)
            current = current.next
            l2 = l2.next
        if carry == 1:
            current.next = ListNode(1)
        return result.next

File: algorithm/2_Add_Two_Numbers/Add_Two_Numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    result := &ListNode{}
    current := result
    carry := 0
    var sum int
    for l1!=nil && l2!=nil{
        sum = l1.Val + l2.Val + carry
        carry = sum / 10
        sum = sum % 10
        current.Next = &ListNode{
            Val : sum,
        }
        current = current.Next
        l1 = l1.Next
        l2 = l2.Next
    }
    for l1!=nil{
        sum = l1.Val + carry
        carry = sum / 10
        sum = sum % 10
        current.Next = &ListNode{
            Val : sum,
        }
        current = current.Next
        l1 = l1.Next
    }
    for l2!=nil{
        sum = l2.Val + carry
        carry = sum / 10
        sum = sum % 10
        current.Next = &ListNode{
            Val : sum,
        }
        current = current.Next
        l2 = l2.Next
    }
    if carry == 1{
        current.Next = &ListNode{
            Val : 1,
        }
    }
    return result.Next
}